toolchain/lexer/string_literal.cpp

```diff
@@ -380,7 +380,7 @@ static auto ExpandEscapeSequencesAndRemoveIndent(
         // Trailing whitespace before a newline doesn't contribute to the string
         // literal value.
         while (!result.empty() && result.back() != '\n' &&
-               IsSpace(result.back())) {
+               result.back() == ' ') {
```

Comment:
What happens for:
```
var s: String = """
\u{0020}
""";
```
? I think in that case the final character of `result` here will be `' '` but we don't want to remove it.

I think we only ever want to remove the characters we just added to `result`. Eg, something like:
```
        while (!result.empty() && result.back() != '\n' &&
               result.back() == ' ' && --end_of_regular_text >= 0) {
```

Reply:
I think the ideal thing here would be to avoid adding anything to `result` until after we've done this skipping backwards step. We could do this scan before the `result += ...` line above, eg:

```
llvm::StringRef regular_text = contents.substr(0, end_of_regular_text);
contents = contents.substr(end_of_regular_text);
if (contents.consume_front("\n")) {
  regular_text = regular_text.rtrim(' ');
}
result += regular_text;
```

Reply:
> I think the ideal thing here would be to avoid adding anything to `result` until after we've done this skipping backwards step.

All of this code is within the `ExpandEscapeSequencesAndRemoveIndent()` method. A string inside Carbon code is being parsed. We shouldn't be removing indents in a Carbon string assignment. Maybe your notes are congruent with my observation. Maybe this method applies to Carbon string assignments and other cases.

Today is my first look at `carbon-lang` code. Bazel and Carbon installed okay on my Ubuntu. I haven't found yet how these cpp tests are executed from explorer using bazel. Current pull request uses `' '` instead of `isspace()` and does appear to fix part of the bigger problem in the original issue.